src/lib.rs

```diff
@@ -97,3 +99,9 @@ pub trait GenFn4<T>: Fn(&PermutationTable, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&PermutationTable, &Point2<T>) -> T, {}
 impl<T, F> GenFn3<T> for F where F: Fn(&PermutationTable, &Point3<T>) -> T, {}
 impl<T, F> GenFn4<T> for F where F: Fn(&PermutationTable, &Point4<T>) -> T, {}
+
+pub trait Module<T> {
+    type Output;
+
+    fn get(&self, point: T) -> Self::Output;
+}
```

Comment:
Wouldn't it be helpful to require that modules be taken by ref, instead of leaving open the possibility of taking them by value, thereby leaving open a possible source of bugs? or should we just expect users to get corrected by rustc where applicable?